Doc/c-api/unicode.rst

```diff
@@ -589,6 +599,7 @@ APIs:
 
    Return a substring of *str*, from character index *start* (included) to
    character index *end* (excluded).  Negative indices are not supported.
+   Returns ``NULL`` and sets an exception on error.
```

Comment:
Returns NULL - is that in all cases? Better to say so and break into short sentences, less confusion more clarity.